packages/backend/src/apps/lettersg/common/add-auth-header.ts

```diff
@@ -1,12 +1,34 @@
 import { TBeforeRequest } from '@plumber/types'
 
+import logger from '@/helpers/logger'
+
 import { getApiBaseUrl } from './api'
 
 const addAuthHeader: TBeforeRequest = async ($, requestConfig) => {
-  const apiKey = $.auth.data.apiKey as string
+  const apiKey = $.auth.data.apiKey
+
+  if (typeof apiKey !== 'string') {
+    logger.error({
+      event: 'auth-header-apikey-error',
+      error: 'LetterSG: API key must be set',
+      flowId: $.flow.id,
+    })
+    throw new Error('Missing API key')
+  }
+
   const baseUrl = getApiBaseUrl(apiKey)
+
+  if (!baseUrl) {
+    logger.error({
+      event: 'auth-header-baseurl-error',
+      error: 'LetterSG: No base URL obtained from API key',
+      flowId: $.flow.id,
+    })
+    throw new Error('No base URL obtained from API key')
+  }
```

Comment:
I think it would be a good idea to add "[LetterSG]" so that we can maintain a greater context of where the error came from.

```suggestion
  if (!baseUrl) {
    logger.error({
      event: 'auth-header-baseurl-error',
      error: '[LetterSG] No base URL obtained from API key',
      flowId: $.flow.id,
    })
    throw new Error('[LetterSG] No base URL obtained from API key')
  }
```